src/gsi/intpm2_5.f90

```diff
@@ -307,6 +308,127 @@ subroutine intpm2_5_(pm2_5head,rval,sval)
 
     end if
 
+!
+   if (laeroana_fv3smoke) then
+       !pm2_5ptr => pm2_5head
+       pm2_5ptr => pm2_5Node_typecast(pm2_5head)
+       do while (associated(pm2_5ptr))
+          j1=pm2_5ptr%ij(1)
+          j2=pm2_5ptr%ij(2)
+          j3=pm2_5ptr%ij(3)
+          j4=pm2_5ptr%ij(4)
+          j5=pm2_5ptr%ij(5)
+          j6=pm2_5ptr%ij(6)
+          j7=pm2_5ptr%ij(7)
+          j8=pm2_5ptr%ij(8)
+          w1=pm2_5ptr%wij(1)
+          w2=pm2_5ptr%wij(2)
+          w3=pm2_5ptr%wij(3)
+          w4=pm2_5ptr%wij(4)
+          w5=pm2_5ptr%wij(5)
+          w6=pm2_5ptr%wij(6)
+          w7=pm2_5ptr%wij(7)
+          w8=pm2_5ptr%wij(8)
+!naero_smoke_fv3
+          iaero=1
+          aeroname=aeronames_smoke_fv3(iaero) !'smoke'
+          call gsi_bundlegetpointer(sval,trim(aeroname),spm2_5,istatus)
```

Comment:
@hongli-wang why don't define the pointers to corresponding sval component out of the loop over observations( loop 315)? Namely move those lines from line 333 out of this loop.

Reply:
Agree. Let's make the code work correct first and then improve the efficiency.